packages/application-components/src/components/dialogs/form-dialog/form-dialog.spec.tsx

```diff
@@ -1,104 +1,95 @@
+      expect(onCancel).toHaveBeenCalled();
+      fireEvent.click(screen.getByLabelText('Save'));
+      expect(onConfirm).toHaveBeenCalled();
+    },
+  });
 };
-DialogController.displayName = 'DialogController';
 
 describe('rendering', () => {
-  it('should open the modal and close it by clicking on the close button', async () => {
-    const onCancel = jest.fn();
-    const onConfirm = jest.fn();
-    renderComponent(
-      <DialogController>
-        {({ isOpen, setIsOpen }) => (
-          <FormDialog
-            title="Lorem ipsus"
-            isOpen={isOpen}
-            onClose={() => setIsOpen(false)}
-            onSecondaryButtonClick={onCancel}
-            onPrimaryButtonClick={onConfirm}
-          >
-            {'Hello'}
-          </FormDialog>
-        )}
-      </DialogController>
-    );
-    expect(screen.queryByText(/Lorem ipsus/i)).not.toBeInTheDocument();
+  let validateComponent: TDialogComponentValidator;
+
+  beforeEach(() => {
+    validateComponent = createFormDialogValidator();
+  });
 
-    fireEvent.click(screen.getByLabelText(/Open Form Dialog/i));
-    await screen.findByText(/Lorem ipsus/i);
-    expect(screen.getByText(/Hello/i)).toBeInTheDocument();
+  it('should open the modal and close it by clicking on the close button', () =>
+    validateComponent({
+      title: 'Lorem ipsus',
+      onClose: (setIsOpen) => setIsOpen(false),
+    }));
 
-    fireEvent.click(screen.getByLabelText('Cancel'));
-    expect(onCancel).toHaveBeenCalled();
-    fireEvent.click(screen.getByLabelText('Save'));
-    expect(onConfirm).toHaveBeenCalled();
+  it('should not be able to close the modal when onClose is not provided', () =>
+    validateComponent({
+      title: 'Lorem ipsus',
+    }));
 
-    fireEvent.click(screen.getByLabelText(/Close dialog/i));
-    expect(screen.queryByText(/Lorem ipsus/i)).not.toBeInTheDocument();
-  });
-  it('should not be able to close the modal when onClose is not provided', async () => {
-    const onCancel = jest.fn();
-    const onConfirm = jest.fn();
-    renderComponent(
-      <DialogController>
-        {({ isOpen }) => (
-          <FormDialog
-            title="Lorem ipsus"
-            isOpen={isOpen}
-            onSecondaryButtonClick={onCancel}
-            onPrimaryButtonClick={onConfirm}
-          >
-            {'Hello'}
-          </FormDialog>
-        )}
-      </DialogController>
-    );
-    expect(screen.queryByText(/Lorem ipsus/i)).not.toBeInTheDocument();
+  it('should show secondaryButton Icon', () =>
+    validateComponent({
+      title: 'Lorem ipsus',
+      extraProps: {
+        iconLeftSecondaryButton: <div>button icon</div>,
+      },
+      extraChecks: () => {
+        screen.getByText('button icon');
+      },
+    }));
+});
 
-    fireEvent.click(screen.getByLabelText(/Open Form Dialog/i));
-    await screen.findByText(/Lorem ipsus/i);
-    expect(screen.getByText(/Hello/i)).toBeInTheDocument();
+describe('with custom title', () => {
```

Comment:
These tests were added to the three dialog components so we verify the new changes work as expected.